OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -0,0 +1,241 @@
+					ownersDropDown.TextColor = selectedOwner.Color.RGB;
+
+					if (currentActor != null) // Hide border of the previously selected actor.
+						currentActor.ShowBorder = false;
+					currentActor = actor;
+					currentActor.ShowBorder = true;
+
+					actorIdField.Text = actor.ID;
+					typeLabel.Text = actor.TypeSummary;
+					actorIdField.CursorPosition = actor.ID.Length;
+					actorSelectBorder.Bounds.Width = actor.Bounds.Width * 2;
+					actorSelectBorder.Bounds.Height = actor.Bounds.Height * 2;
+					HideValidationMessage();
+				}
+
+				actorSelectBorder.Bounds.X = origin.X;
+				actorSelectBorder.Bounds.Y = origin.Y;
+
+				// Set the edit panel to the right of the selection border.
+				int newX = origin.X + actorSelectBorder.Bounds.Width + editPanelPadding;
+				actorEditPanel.Bounds.X = newX;
+				actorEditPanel.Bounds.Y = origin.Y;
+			}
+			else
+			{
+				// Selected actor is null, hide the border and edit panel.
+				if (currentActor != null)
+					currentActor.ShowBorder = false;
+
+				actorIdField.YieldKeyboardFocus();
+				currentActor = null;
+			}
+		}
+
+		void ShowValidationMessage(string message)
+		{
+			actorEditPanel.Bounds.Height = validationHeight;
+			validationLabel.Text = message;
+			validationLabel.Visible = true;
+			actorIdField.TakeKeyboardFocus();
+		}
+
+		void HideValidationMessage()
+		{
+			actorEditPanel.Bounds.Height = originalHeight;
+			validationLabel.Visible = false;
+		}
+
+		void Save()
+		{
+			if (currentActor == null)
+				return;
+
+			var newActorName = actorIdField.Text;
+			var newActorNameLower = newActorName.ToLowerInvariant();
+
+			if (string.IsNullOrWhiteSpace(newActorName))
+			{
+				ShowValidationMessage("Please enter an actor ID.");
+				return;
+			}
+
+			// Check for duplicate actor ID
+			if (currentActor.ID.ToLowerInvariant() != newActorNameLower)
+			{
+				bool found = world.Map.ActorDefinitions.Any(x => x.Key.ToLowerInvariant() == newActorNameLower);
+				if (found)
+				{
+					ShowValidationMessage("An actor is already using this ID.");
+					return;
+				}
+			}
+
+			// Change key in actor defs
+			var actorDef = world.Map.ActorDefinitions.First(x => x.Key == currentActor.ID);
+			actorDef.Key = newActorName;
+
+			// Remove existing editor actor
+			editorActorLayer.Remove(currentActor);
+
+			// Update the owner
+			var actorDict = actorDef.Value.ToDictionary();
+			actorDict["Owner"].Value = selectedOwner.Name;
+
+			// Create the new actor reference and add to the layer. Get a new EditorActorPreview.
+			var newActorReference = new ActorReference(actorDef.Value.Value, actorDict);
+			currentActor = editorActorLayer.Add(newActorName, newActorReference, false);
+			currentActor.ShowBorder = true;
+
+			editor.DefaultBrush.SelectedActor = currentActor; // Update the selection brush's selected actor.
+
+			HideValidationMessage();
+		}
+
+		void Delete()
+		{
+			if (currentActor != null)
+				editorActorLayer.Remove(currentActor);
+			Cancel();
```

Comment:
General comment: try and be consistent about newlines after if statements (prefer having them).